sycl/source/detail/scheduler/commands.cpp

```diff
@@ -203,17 +203,34 @@ static std::string commandToName(Command::CommandType Type) {
 }
 #endif
 
-static std::vector<RT::PiEvent>
-getPiEvents(const std::vector<EventImplPtr> &EventImpls) {
+std::vector<RT::PiEvent>
+Command::getPiEvents(const std::vector<EventImplPtr> &EventImpls) const {
   std::vector<RT::PiEvent> RetPiEvents;
   for (auto &EventImpl : EventImpls) {
-    if (EventImpl->getHandleRef() != nullptr)
-      RetPiEvents.push_back(EventImpl->getHandleRef());
+    if (EventImpl->getHandleRef() == nullptr)
+      continue;
+
+    // Do not add redundant event dependencies for in-order queues.
+    // At this stage dependency is definitely pi task and need to check if
+    // current one is a host task. In this case we should not skip pi event due
+    // to different sync mechanisms for different task types on in-order queue.
+    const auto &WorkerQueue = getWorkerQueue();
```

Comment:
```suggestion
    const QueueImplPtr &WorkerQueue = getWorkerQueue();
```

Reply:
fixed in https://github.com/intel/llvm/pull/5901/commits/7d5def94400fe92b178f43bcc778a0bd9b00d9b6